### DISCUSSION TITLE
Is it possible to render nothing when render function return nil

### DISCUSSION THREAD

**Discussion by fitrh** [b0o/incline.nvim#16] [Questions and Answers] [answered]
For example, I have something like this in my render function

```lua
  if props.focused then
    local ok, gps = pcall(require, "nvim-gps")
    if not ok or not gps.is_available() or (gps.get_location() == "") then
      return nil
    end

    return { gps.get_location() }
  end
```

The result

> I still get a little box when the result is nil

https://user-images.githubusercontent.com/55179750/168426413-453c2308-b5ae-48dd-93ce-2af6847bfdab.mp4

Is it possible to completely get rid of the render result when we return `nil` ?

**Comment by syphar** [comment]
I had the same issue when I wanted to hide incline when my cursor is on the same line, so it doesn't hide source. 

My current workaround is an autocmd: 
```lua
vim.api.nvim_create_autocmd({ "CursorMoved", "CursorMovedI" }, {
	group = vim.api.nvim_create_augroup("hide_incline_on_first_line", {}),
	buffer = 0,
	callback = function()
		local incline = require("incline")
		if vim.fn.line(".", vim.api.nvim_get_current_win()) == 1 then
			incline.disable()
		else
			incline.enable()
		end
	end,
})
```

There is probably a better way to do this. 

( I also probably will  start digging into incline itself for a PR, but that takes more time)

**Reply by b0o** [reply]
In addition to fitrh’s suggestion, I plan to add a configuration option to hide the incline window if the cursor is on the same line.

**Reply by syphar** [reply]
oh, that would be awesome! ❤️

**Comment by b0o** [comment, marked as answer]
~~This isn’t currently possible but it’s a great idea. I will implement it soon.~~

**Update:** 

This is now implemented. If `render` returns `nil`, that statusline is temporarily hidden until `render` returns non-nil. See `:help incline-config-render`.

Also added `config.hide.cursorline`. See `:help incline-config-hide.cursorline`.

**Comment by b0o** [comment]
By the way @fitrh I really like your idea of using Incline to display gps info. Would you like to contribute a screenshot/screencast to the [showcase](https://github.com/b0o/incline.nvim/discussions/categories/showcase)? And if so, would it be okay if I add it to the README?

**Reply by fitrh** [reply]
Sure, I will do it when I have time

**Comment by b0o** [comment]
This has been added, see my [updated comment](https://github.com/b0o/incline.nvim/discussions/16#discussioncomment-2752074).
